Apply current development practices to this role

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request updates this role to ensure it is following our current development practices. This includes:
- Using fully-qualified collection names
- Alphabetizing task arguments
- Ensuring appropriate separation of testing requirements from role functionality
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

I am hunting down a deprecation warning while building fresh [CyHy AMIs](https://github.com/cisagov/cyhy_amis) and this is the first Ansible dependency I checked. While the warning was not found here I saw several out-of-date items and thought I would correct them while I'm here.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass successfully.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

mcdonnnj committed:
Add explicit YAML start of document

We use these in all our YAML files as standard practice now. Also
remove an unnecessary comment.

mcdonnnj committed:
Use FQCNs for tasks

Ansible strongly encourages, and we have switched to, using fully-
qualified collection names for tasks.

mcdonnnj committed:
Sort task arguments alphabetically

This has become our standard practice and so it is applied here.

mcdonnnj committed:
Move sshd installation to the molecule configuration

Based on the comment preceding the task and what this role is described
as doing the task to install openssh-server does not belong in the
role's tasks. I have moved it to the molecule prepare configuration as
that most closely aligned with the explanatory comment.

dav3r reviewed (approved):
👍

jsf9k commented on molecule/default/prepare.yml:
```diff
@@ -1,3 +1,14 @@
 ---
 - name: Import upgrade playbook
   ansible.builtin.import_playbook: upgrade.yml
+
+# This is already installed on the AMIs we use, but not on the Docker
+# images we use for our molecule tests
+- hosts: all
+  name: Ensure that sshd is installed
+  become: yes
+  become_method: sudo
+  tasks:
+    - name: Install openssh-server
+      ansible.builtin.package:
+        name: openssh-server
```
I don't understand the point of moving this installation to the prepare playbook.  We must install sshd so that `/etc/ssh/sshd_config` exists for us to modify.  That is why I think the original location of `tasks/main.yml` is a better fit for the installation of `openssh-server`.

mcdonnnj replied:
Then this role needs to be updated to otherwise mention that it's doing that. I would never assume that `openssh-server` is installed from all other documentation in this role. The comment for that particular task even makes it sound like it's only there for testing.

jsf9k replied:
Please see commits bad3f96 and 8ab66d3.  Feel free to remove 5bc06c8 and bad3f96 and force push to clean up the history.

You may also go ahead and resolve this conversation if you are satisfied with the changes I made.

mcdonnnj replied:
@jsf9k I'm going to leave the commit history for record keeping. I would still like the OpenSSH installation broken out at some point because it's a weird (albeit necessary right now) fit for this role.

jsf9k committed:
Revert "Move sshd installation to the molecule configuration"

This reverts commit 5bc06c8543e008360568a39a0b498253e91d7cb7.

jsf9k committed:
Update README to mention that this role installs OpenSSH server

Also update the corresponding comment in tasks.main.yml to make it
clearer that this is an expected behavior of the role.

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

jsf9k commented:
@mcdonnnj - What is the deprecation warning you are hunting down?

mcdonnnj commented:
> @mcdonnnj - What is the deprecation warning you are hunting down?

@jsf9k Ah I found the source yesterday in https://github.com/GekoCloud/ansible-role-disable-thp. They still use an `include` instead of one of the more specific modules and that will be killed in Ansible 2.16. I will probably just roll our own equivalent role after I finish up this other CyHy work as they haven't released Ansible 2.15 yet so there's time before it potentially blows up in our faces.

jsf9k commented:
> > @mcdonnnj - What is the deprecation warning you are hunting down?
> 
> @jsf9k Ah I found the source yesterday in https://github.com/GekoCloud/ansible-role-disable-thp. They still use an `include` instead of one of the more specific modules and that will be killed in Ansible 2.16. I will probably just roll our own equivalent role after I finish up this other CyHy work as they haven't released Ansible 2.15 yet so there's time before it potentially blows up in our faces.

@mcdonnnj - are you able to approve this one now and resolve conversations, or is there something more I need to address?

jsf9k commented:
> > > @mcdonnnj - What is the deprecation warning you are hunting down?
> > 
> > 
> > @jsf9k Ah I found the source yesterday in https://github.com/GekoCloud/ansible-role-disable-thp. They still use an `include` instead of one of the more specific modules and that will be killed in Ansible 2.16. I will probably just roll our own equivalent role after I finish up this other CyHy work as they haven't released Ansible 2.15 yet so there's time before it potentially blows up in our faces.
> 
> @mcdonnnj - are you able to approve this one now and resolve conversations, or is there something more I need to address?

@mcdonnnj - Bumping this for visibility.

mcdonnnj committed:
Update the README

Use the GitHub-flavored Markdown syntax for a note. Also capitalize SSH
to correctly reflect preferred case when referring to Secure Shell.

jsf9k commented:
@mcdonnnj, so do I have your approval for this PR?

jsf9k reviewed (approved):
_I LOVE IT!!!_
